pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -84,16 +86,186 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
 			}
 		},
 		func(obj *LeaderElectionConfiguration) {
-			zero := unversioned.Duration{}
-			if obj.LeaseDuration == zero {
+			if obj.LeaseDuration == zeroDuration {
 				obj.LeaseDuration = unversioned.Duration{15 * time.Second}
 			}
-			if obj.RenewDeadline == zero {
+			if obj.RenewDeadline == zeroDuration {
 				obj.RenewDeadline = unversioned.Duration{10 * time.Second}
 			}
-			if obj.RetryPeriod == zero {
+			if obj.RetryPeriod == zeroDuration {
 				obj.RetryPeriod = unversioned.Duration{2 * time.Second}
 			}
 		},
+		func(obj *KubeletConfiguration) {
+			if obj.Address == "" {
+				obj.Address = "0.0.0.0"
+			}
+			if obj.CAdvisorPort == 0 {
+				obj.CAdvisorPort = 4194
+			}
+			if obj.CertDirectory == "" {
+				obj.CertDirectory = "/var/run/kubernetes"
+			}
+			if obj.ConfigureCBR0 == nil {
+				obj.ConfigureCBR0 = boolVar(false)
+			}
+			if obj.ContainerRuntime == "" {
+				obj.ContainerRuntime = "docker"
+			}
+			if obj.CPUCFSQuota == nil {
+				obj.CPUCFSQuota = boolVar(true)
+			}
+			if obj.DockerExecHandlerName == "" {
+				obj.DockerExecHandlerName = "native"
+			}
+			if obj.DockerEndpoint == "" {
+				obj.DockerEndpoint = "unix:///var/run/docker.sock"
+			}
+			if obj.EventBurst == 0 {
+				obj.EventBurst = 10
+			}
+			if obj.EventRecordQPS == 0 {
```

Comment:
This is not consistent with the documentation of this field:
> If 0, there is no limit enforced.

This is why we need pointers for primitives...